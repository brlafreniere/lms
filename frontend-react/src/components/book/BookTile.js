import React from "react";

import {Link} from "react-router-dom";

export default function BookTile(props) {
    return (
        <Link to={`/books/${props.book.id}`}>
            <div className="book-tile" style={{'backgroundImage': `url(${process.env.REACT_APP_API_URL}/uploads/${props.book.cover_image_file_name})`}}>
                <div className='book-tile-title'>
                    {props.book.title}
                </div>
            </div>
        </Link>
    )
}