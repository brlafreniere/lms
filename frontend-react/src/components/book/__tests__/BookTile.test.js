import React from "react";
import BookTile from "../BookTile";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

test("snapshot test", () => {
    let book = {
        id: 2,
        cover_image_file_name: "foo.jpg",
        title: "foo title"
    };
    const component = renderer.create(
        <MemoryRouter>
            <BookTile book={book} />
        </MemoryRouter>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})