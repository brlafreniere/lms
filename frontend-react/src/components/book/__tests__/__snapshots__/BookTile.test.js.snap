// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<a
  href="/book/2"
  onClick={[Function]}
>
  <div
    className="book-tile"
    style={
      Object {
        "backgroundImage": "url(undefined/uploads/foo.jpg)",
      }
    }
  >
    <div
      className="book-tile-title"
    >
      foo title
    </div>
  </div>
</a>
`;
